src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -3,13 +3,241 @@ Tests for external textures from HTMLVideoElement (and other video-type sources?
+          module: t.device.createShaderModule({
+            code: `
+              [[stage(vertex)]] fn main([[builtin(vertex_index)]] VertexIndex : u32) -> [[builtin(position)]] vec4<f32> {
+                  var pos = array<vec4<f32>, 6>(
+                    vec4<f32>( 1.0,  1.0, 0.0, 1.0),
+                    vec4<f32>( 1.0, -1.0, 0.0, 1.0),
+                    vec4<f32>(-1.0, -1.0, 0.0, 1.0),
+                    vec4<f32>( 1.0,  1.0, 0.0, 1.0),
+                    vec4<f32>(-1.0, -1.0, 0.0, 1.0),
+                    vec4<f32>(-1.0,  1.0, 0.0, 1.0)
+                  );
+                  return pos[VertexIndex];
+              }
+              `,
+          }),
+          entryPoint: 'main',
+        },
+        fragment: {
+          module: t.device.createShaderModule({
+            code: `
+              [[group(0), binding(0)]] var s : sampler;
+              [[group(0), binding(1)]] var t : texture_external;
+
+              [[stage(fragment)]] fn main([[builtin(position)]] FragCoord : vec4<f32>)
+                                       -> [[location(0)]] vec4<f32> {
+                  return textureSampleLevel(t, s, FragCoord.xy / vec2<f32>(16.0, 16.0));
+              }
+              `,
+          }),
+          entryPoint: 'main',
+          targets: [
+            {
+              format: kFormat,
+            },
+          ],
+        },
+        primitive: { topology: 'triangle-list' },
+      });
+
+      const linearSampler = t.device.createSampler();
+
+      const externalTextureDescriptor = { source: video };
+      const externalTexture = t.device.importExternalTexture(externalTextureDescriptor);
+
+      const bindGroup = t.device.createBindGroup({
+        layout: pipeline.getBindGroupLayout(0),
+        entries: [
+          {
+            binding: 0,
+            resource: linearSampler,
+          },
+          {
+            binding: 1,
+            resource: externalTexture,
+          },
+        ],
+      });
+      const commandEncoder = t.device.createCommandEncoder();
+      const passEncoder = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: colorAttachment.createView(),
+            loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+            storeOp: 'store',
+          },
+        ],
+      });
+      passEncoder.setPipeline(pipeline);
+      passEncoder.setBindGroup(0, bindGroup);
+      passEncoder.draw(6);
+      passEncoder.endPass();
+      t.device.queue.submit([commandEncoder.finish()]);
+
+      // Top left corner should be red. Sample a few pixels away from the edges to avoid color
+      // management artifacts.
+      t.expectSinglePixelIn2DTexture(
+        colorAttachment,
+        kFormat,
+        { x: 2, y: 2 },
+        {
+          exp: new Uint8Array([0xff, 0x00, 0x00, 0xff]),
+        }
+      );
+
+      // Bottom right corner should be green. Sample a few pixels away from the edges to avoid color
+      // management artifacts.
```

Comment:
nit: I think if there are corner artifacts it would be due to compression, not color management
```suggestion
      // Bottom right corner should be green. Sample a few pixels away from the edges to avoid
      // compression artifacts.
```

Reply:
Done.